plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2567,10 +2567,10 @@ private static HyperLogLog getHyperLogLogForNdv(Block block)
 
     private static Optional<Instant> getMaxFileModificationTime(Collection<ComputedStatistics> computedStatistics)
     {
-        // Only statistics for whole table are collected
-        ComputedStatistics singleStatistics = Iterables.getOnlyElement(computedStatistics);
-
-        return singleStatistics.getColumnStatistics().entrySet().stream()
+        return computedStatistics.stream()
+                .map(ComputedStatistics::getColumnStatistics)
+                .map(Map::entrySet)
+                .flatMap(Collection::stream)
                 .filter(entry -> entry.getKey().getColumnName().equals(FILE_MODIFIED_TIME_COLUMN_NAME))
```

Comment:
Grouping is defined for whole table, so each columns will have grouping (including `FILE_MODIFIED_TIME_COLUMN_NAME`). In case of grouping by $path in following commit we receive $file_modified_time for each file and calculate max value.